x/bridge/observer/observer.go

```diff
@@ -116,31 +139,69 @@ func (o *Observer) sendOutbound(ctx context.Context) {
 
 	for srcId, queue := range o.outTxQueue {
 		srcChain := o.chains[srcId]
-		height, err := srcChain.Height()
+		height, err := srcChain.Height(ctx)
 		if err != nil {
 			o.logger.Error(fmt.Sprintf("Failed to get height for %s: %s", srcId, err.Error()))
 			continue
 		}
-		confirmationsRequired, err := srcChain.ConfirmationsRequired()
-		if err != nil {
-			o.logger.Error(fmt.Sprintf(
-				"Failed to get confirmations required for %s: %s",
-				srcId,
-				err.Error(),
-			))
-			continue
-		}
-		var newQueue []Transfer
+
+		var newQueue []TxQueueItem
 		for _, out := range queue {
-			if height-out.Height < confirmationsRequired {
+			confirmed, err := o.isTxConfirmed(ctx, height, &out)
+			if !confirmed || err != nil {
+				if err != nil {
+					o.logger.Error(fmt.Sprintf("Failed to confirm tx %s", err.Error()))
+				}
 				newQueue = append(newQueue, out)
 				continue
 			}
-			err = o.chains[out.DstChain].SignalInboundTransfer(ctx, out)
+			err = o.chains[out.Tx.DstChain].SignalInboundTransfer(ctx, out.Tx)
 			if err != nil {
 				newQueue = append(newQueue, out)
 			}
 		}
 		o.outTxQueue[srcId] = newQueue
 	}
 }
+
+func (o *Observer) getConfirmationsRequired(ctx context.Context, tx Transfer) (uint64, error) {
+	if tx.SrcChain == ChainIdOsmosis {
+		return 0, nil
+	}
```

Comment:
Question on this, we still would need 1 confirmation for Osmosis right?

Reply:
Hm okay, I guess as long as we define confirmations as "number of blocks after the transaction is in a block" then this works. But for bitcoin, I think when a transaction is first in a block that counts as 1 confirmation right? We shouldnt have differing definitions of what a confirmation is

Reply:
In Bitcoin, the number of confirmations: `currentHeight - txHeight + 1`, so yes, the first block counts as 1. In that case we need to modify the conditions in the `isTxConfirmed` method to have
```go
	if curHeight - item.Tx.Height + 1 < item.ConfirmationsRequired {
		return false, nil
	}
```
And return 1 here

Reply:
Fixed!